documentation/source/KD.md

```diff
@@ -0,0 +1,119 @@
+
+Knowledge distillation is a technique in deep learning that aims to transfer the knowledge of a large, pre-trained neural network model (the "teacher") to a smaller, more computationally efficient model (the "student"). This is accomplished by training the student to mimic the predictions of the teacher, in addition to the ground-truth labels. The student network can also be designed to have a different architecture from the teacher, making it possible to distill the knowledge of a complex teacher network into a lighter and faster student network for deployment in real-world applications.
+
+
+The flow of training with Knowledge distillaiton in SG is similar to regular training. For regular training, we used SGs `Trainer` class - which was charge of training the model, evaluating test data, making predictions, and saving checkpoints.
+Equivalently, for knowledge distillation we use the `KDTrainer` class which inherits from `Trainer`.
+
+If for regular training with `Trainer`, the general flow is:
+
+```python
+...
+trainer = Trainer("my_experiment")
+train_dataloader = ...
+valid_dataloader = ...
+model = ...
+
+train_params = {...}
+trainer.train(model=model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+```
+
+
+Then for training with knowledge distillation, the general flow is:
+```python
+
+from super_gradients.training.kd_trainer import KDTrainer
+...
+kd_trainer = KDTrainer("my_experiment")
+train_dataloader = ...
+valid_dataloader = ...
+
+student_model = ...
+teacher_model = ...
+
+train_params = {...}
+kd_trainer.train(student=student_model, teacher=teacher_model, training_params=train_params, train_loader=train_dataloader, valid_loader=valid_dataloader)
+```
+
+Check out our [knowledge distillation tutorial notebook](https://bit.ly/3BLA5oR) to see a practical example in action.
+
+
+## Knowledge Distillation Training in SG: Key Components
+
+
+### [KDModule](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/kd_modules/kd_module.py)
+
+The most apparent difference in the flow of training using knowledge distillation is the fact that it requires two networks: the "teacher" and the "student".
+The relation between the two is also configurable - for example, we may decide that the teacher model should preprocess the inputs differently.
+For that matter, SG introduces a new `torch.nn.Module` that wraps both the student and the teacher models: `KDModule`.
+Uppon calling `KDTrainer.train()` the teacher and student models are passed, along the `kd_arch_params` to initialize a `KDModule` instance.
+Passing a `KDModule` instance explicitly to `KDTrainer.train()` through the `model` argument instead of student and teacher models is also possible, which gives the users the option to customize KD to their needs.
```

Comment:
it is not so clear what can we use the KDModule for. 
maybe add an example

Reply:
So...currently, there is no use for it since we only apply KD directly to the student and teacher outputs.
This was my design proposal so that in the future, we can customise KD (for example, define a different KDModule that might extract additional intermediate layers) and inherit from it.

Reply:
Added high-level customisation example.